### PR TITLE
Handle cases where `CustomEvent` was not initially sent

### PR DESCRIPTION
**Pull request by JamesWTruher**
Since the CustomEvent is sent only on startup, we need a way to send
the startup event in those cases where PowerShell is being hosted
(not the console). This will check to see if we have already sent the
custom event for startup and send if it hasn't. This will enable us
to track some of the hosted scenarios like Jupyter.

# PR Summary

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 1]
```diff
@@ -87,6 +87,9 @@ public static class ApplicationInsightsTelemetry
         // the session identifier
         private static string s_sessionId { get; set; }
 
+        // private semaphore to determine whether we sent the startup telemetry event
+        private static bool s_startupEventSent { get; set; }
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/utils/Telemetry.cs]: should this just be a field?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 2]
```diff
@@ -87,6 +87,9 @@ public static class ApplicationInsightsTelemetry
         // the session identifier
```
Comment (PoshChan) [inline src/System.Management.Automation/utils/Telemetry.cs]: @JamesWTruher, your last commit had 2 failures in `PowerShell-CI-linux`
CustomEvent resend check.Should resend startup event if the semaphore says we haven't sent telemetry
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Basic/Telemetry.Tests.ps1: line 14
14:         $telemetrySent | Should -Be $true
```
CustomEvent resend check.Should resend startup event if the semaphore says we haven't sent telemetry
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Basic/Telemetry.Tests.ps1: line 14
14:         $telemetrySent | Should -Be $true
```

Comment (PoshChan) [inline src/System.Management.Automation/utils/Telemetry.cs]: @JamesWTruher, your last commit had 2 failures in `PowerShell-CI-windows`
CustomEvent resend check.Should resend startup event if the semaphore says we haven't sent telemetry
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\Telemetry.Tests.ps1: line 14
14:         $telemetrySent | Should -Be $true
```
CustomEvent resend check.Should resend startup event if the semaphore says we haven't sent telemetry
```powershell
Expected $true, but got $false.
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\Telemetry.Tests.ps1: line 14
14:         $telemetrySent | Should -Be $true
```

Comment (PoshChan) [inline src/System.Management.Automation/utils/Telemetry.cs]: @JamesWTruher, your last commit had 2 failures in `PowerShell-CI-macos`
CustomEvent resend check.Should resend startup event if the semaphore says we haven't sent telemetry
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /Users/runner/runners/2.164.7/work/1/s/test/powershell/engine/Basic/Telemetry.Tests.ps1: line 14
14:         $telemetrySent | Should -Be $true
```
CustomEvent resend check.Should resend startup event if the semaphore says we haven't sent telemetry
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /Users/runner/runners/2.164.7/work/1/s/test/powershell/engine/Basic/Telemetry.Tests.ps1: line 14
14:         $telemetrySent | Should -Be $true
```

File: src/System.Management.Automation/utils/Telemetry.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/utils/Telemetry.cs]: Looks like the other members are all properties.  We can decide to change them as fields in a separate PR if needed.

**Review by SteveL-MSFT — APPROVED** [review]

File: src/System.Management.Automation/utils/Telemetry.cs [hunk 3]
```diff
@@ -666,6 +676,7 @@ internal static void SendPSCoreStartupTelemetry(string mode)
             try
             {
                 s_telemetryClient.TrackEvent("ConsoleHostStartup", properties, null);
+                s_startupEventSent = true;
```
Comment (TylerLeonhardt) [inline src/System.Management.Automation/utils/Telemetry.cs]: Should we be concerned about thread safety here?

**Comment by TravisEz13** [comment]
@JamesWTruher I'm going to address the comments because your are OOF and we need this for GA.

**Review by daxian-dbw — APPROVED** [review]

File: src/System.Management.Automation/utils/Telemetry.cs (hunk 3 above)
Comment (TravisEz13) [reply src/System.Management.Automation/utils/Telemetry.cs]: Addressed

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 hour ago

**Review by JamesWTruher — COMMENTED** [review]
lgtm

**Pull request merged**